File: goal-parser-interpretation/goal-parser-interpretation.cpp
class Solution {
public:
    string interpret(string command) {
        
        std::string goal = "";
        
        int i = 0;
        
        while (i < command.size()) {
            
            if (command[i++] == '(') {
                
                if (command[i++] == ')') {
                    
                    goal += 'o';
                }
                else {
                    
                    goal += "al";
                    
                    i += 2;
                }
            }
            else {
                
                goal += "G";
            }
        }
        
        return goal;
    }
};

File: goal-parser-interpretation/goal-parser-interpretation.java
class Solution {
    public String interpret(String command) {
        
        StringBuilder goal = new StringBuilder();
        
        int i = 0;
        
        while (i < command.length()) {
            
            if (command.charAt(i++) == '(') {
                
                if (command.charAt(i++) == ')') {
                    
                    goal.append('o');
                }
                else {
                    
                    goal.append("al");
                    
                    i += 2;
                }
            }
            else {
                
                goal.append('G');
            }
        }
        
        return goal.toString();
    }
}